frontend/src/store/FamLoginUserState.ts

```diff
@@ -0,0 +1,168 @@
+import type { CognitoUserSession } from 'amazon-cognito-identity-js';
+import { AdminRoleAuthGroup, type FamApplicationDto, type FamAuthGrantDto } from 'fam-admin-mgmt-api/model';
+import { readonly, ref } from "vue";
+
+const FAM_LOGIN_USER = 'famLoginUser';
+
+// For some of the values of the FAMLoginUser properties, see AuthService.parseToken().
+export interface FamLoginUser {
+    username?: string;
+    displayName?: string;
+    email?: string;
+    idpProvider?: string; // from ID Token's ['identities']['providerName'] attribute.
+    roles?: string[]; // roles from Access Token's ['cognito:groups']. This may soon be redundant after delegated admin design.
+    authToken?: CognitoUserSession; // original JWT token from AWS Cognito (ID && Access Tokens).
+    accesses?: FamAuthGrantDto[]; // admin privileges retrieved from backend.
+};
+
+const state = ref({
+    famLoginUser: localStorage.getItem(FAM_LOGIN_USER)
+        ? (JSON.parse(localStorage.getItem(FAM_LOGIN_USER) as string) as
+              | FamLoginUser
+              | undefined
+              | null)
+        : undefined,
+});
+
+// --- getters
+
+const getAuthToken = () => {
+    return state.value.famLoginUser?.authToken;
+};
+
+const getUserAccess = () => {
+    return state.value.famLoginUser?.accesses;
+};
+
+// admin levels: 'FAM_ADMIN', 'APP_ADMIN', 'DELEGATED_ADMIN'
+const getUserAdminRoleGroups = () => {
+    const accesses = getUserAccess();
+    if (accesses && accesses.length > 0) {
+        return accesses.map(
+            (access) => access.auth_key
+        )
+    }
+};
+
+/**
+ * @param adminRole three admin levels: 'FAM_ADMIN', 'APP_ADMIN', 'DELEGATED_ADMIN'
+ * @returns array of applications the login user has been granted for
+ *          administering on based on adimn level.
+ */
+const getApplicationsAdministeredByAdminRole = (
+    adminRole: AdminRoleAuthGroup
+): FamApplicationDto[] | undefined => {
+    const accesses = getUserAccess();
+    if (accesses && accesses.length > 0) {
+        const access = accesses.filter(
+            (access) => access.auth_key == adminRole
+        )[0]
+        return access
+            .grants
+            .map((grant) => grant.application)
+            .sort((first, second) => first.id - second.id);
+    }
+};
+
+/**
+ * Note!! this is to combine all applications the user has been granted
+ * and can administer on with different admin level privileges, only for
+ * special usage at dashboard page (ManagePermissions) since the user when
+ * it lands on the first page it needs to select "application" first.
+ *
+ * If the way our forntend works is based on admin level to display
+ * management table instead of selecting "applicatoin" first, it won't need
+ * this particular function.
+ */
+const getApplicationsUserAdministers = () => {
+    const userAdminRoleGroups = getUserAdminRoleGroups();
+    let famApp;
+    const applicationList: Array<FamApplicationDto> = [];
+    if (userAdminRoleGroups) {
```

Comment:
so maybe we can just go through the access array instead of go through the userAdminRoleGroups and call `getApplicationsAdministeredByAdminRole`, I feel it's a little bit hard to follow

Reply:
Good suggestion, although I was reusing other function for convenience but the extra mapping is unnecessary. the logic refactored.